Toutes les traductions de la documentation pour la v2.6 (#185)
vuejs-fr/vuejs.org: src/v2/api/index.md

  > En 2.4.0+ ce hook capture également les erreurs lancées depuis un gestionnaire d'évènement Vue personnalisé.

>   > En 2.6.0+, ce hook capture également les erreurs lancées depuis les écouteurs de DOM `v-on`. De plus, si les hooks couverts ou gestionnaires retournent une chaine de Promesse, (par exemple les fonctions async), l'erreur sera également remontée depuis cette Promesse.

Reviewer: ```suggestion
  > En 2.6.0+, ce hook capture également les erreurs lancées depuis les écouteurs de DOM `v-on`. De plus, si les hooks couverts ou gestionnaires retournent une chaine de Promesse (par exemple les fonctions async), l'erreur sera également remontée depuis cette Promesse.
```

Double virgule inutile

- **Voir aussi :** [Fonctions de rendu](../guide/render-function.html)

### Vue.observable( object )

> Nouveau en 2.6.0+

- **Arguments :**
  - `{Object} object`

- **Utilisation :**

>   Rend un objet réactif. En interne, Vue utilise cela sur l'objet retourné par la fonction `data`.

Reviewer 2: Que penses-tu de : 

```suggestion
  Rend un objet réactif. En interne, Vue l'utilise sur l'objet retourné par la fonction `data`.
```

>   L'objet retourné peut être directement à l'intérieur des [fonctions de rendu](../guide/render-function.html) et [propriétés calculées](../guide/computed.html) et déclenchera les mises à jour appropriées lors de mutations. Il peut également être utilisé comme un store minimal inter-composant dans des scénarios basiques :

Reviewer 2: > être directement **utilisé** à l'intérieur

Reviewer 2: Peut-être placer une virgule entre `[propriétés calculées](../guide/computed.html)` et `et déclenchera` pour faire une petite pause ? 

On enchaine deux `et`, je trouve pas ça tip top à la lecture :thinking:

Author: Bien vu !

J'ai remplacé le premier par ainsi que. J'ai fini par mettre un point et reprendre une phrase avec il.

>   L'objet retourné peut être directement utilisé à l'intérieur des [fonctions de rendu](../guide/render-function.html) ainsi que des [propriétés calculées](../guide/computed.html). Il déclenchera les mises à jour appropriées lors de mutations. Il peut également être utilisé comme un store minimal inter-composant dans des scénarios basiques :

Reviewer 3: - intercomposant (jamais de trait d'union pour les mots avec le préfixe inter)

Author: « (jamais de trait d'union pour les mots avec le préfixe inter) »

Que ceci reste graver à jamais dans ma mémoire !

    render(h) {
      return h('button', {
        on: { click: () => { state.count++ }}
      }, `le compteur est : ${state.count}`)
    }
  }
  ```

>   <p class="tip">Dans Vue 2.x, `Vue.observable` mute directement l'objet qui lui est transmis, de sorte qu'il soit équivalent à l'objet renvoyé, comme [démontré ici](../guide/instance.html#Donnees-et-methodes). Dans Vue 3.x, un proxy réactif sera retourné à la place, laissant l'objet original non réactif s'il est muté directement. Cependant, pour de future compatibilité, nous recommandons toujours de travailler avec l'objet retourné par `Vue.observable`, au lieu de l'objet originel.</p>

Reviewer 3: - pour de futures compatibilités

>   <p class="tip">Dans Vue 2.x, `Vue.observable` mute directement l'objet qui lui est passé, ainsi cela est équivalent à l'objet retourné, comme [démontré ici](../guide/instance.html#Donnees-et-methodes). Dans Vue 3.x, un proxy réactif sera retourné à la place, laissant l'objet original non réactif s'il est muté directement. Cependant, pour de future compatibilité, nous recommandons toujours de travailler avec l'objet retourné par `Vue.observable`, au lieu de l'objet originalement passé.</p>

Reviewer: ```suggestion
  <p class="tip">Dans Vue 2.x, `Vue.observable` mute directement l'objet qui lui est passé, ainsi cela est équivalent à l'objet retourné, comme [démontré ici](../guide/instance.html#Donnees-et-methodes). Dans Vue 3.x, un proxy réactif sera retourné à la place, laissant l'objet original non réactif s'il est muté directement. Cependant, pour de future compatibilité, nous recommandons toujours de travailler avec l'objet retourné par `Vue.observable`, au lieu de l'objet originel.</p>
```

Suggestion…

>   <p class="tip">Dans Vue 2.x, `Vue.observable` mute directement l'objet qui lui est passé, ainsi cela est équivalent à l'objet retourné, comme [démontré ici](../guide/instance.html#Donnees-et-methodes). Dans Vue 3.x, un proxy réactif sera retourné à la place, laissant l'objet original non réactif s'il est muté directement. Cependant, pour de future compatibilité, nous recommandons toujours de travailler avec l'objet retourné par `Vue.observable`, au lieu de l'objet originalement passé.</p>

Reviewer 2: > ... mute directement l'objet qui lui est passé, ainsi cela est équivalent à l'objet retourné ...

je trouve qu'il y a quelque chose qui va pas dans cette phrase, mais je ne saurais pas dire... :confused:

Reviewer 4: **une proposition :**
Dans Vue 2.x, `Vue.observable` mute directement l'objet qui lui est transmis, de sorte qu'il soit équivalent à l'objet renvoyé, ...

> - **Voir aussi :** [Réactivité dans le détail](../guide/reactivity.html)

Reviewer 3: Je mettrais "Réactivité en détail"

  Une liste ou un objet décrivant les attributs exposés par le composant afin de passer des données depuis le composant parent. Ce paramètre a une syntaxe simple basée sur un tableau (`Array`) et une syntaxe alternative basée sur un objet (`Object`) qui permet une configuration avancée telle qu'une vérification de typage, des contrôles de validation personnalisés et des valeurs par défaut.

  Avec la syntaxe objet, vous pouvez passer les options suivantes :

>     - Le **type** peut être l'un des constructeurs natif suivant : `String`, `Number`, `Boolean`, `Array`, `Object`, `Date`, `Function`, `Symbol` ainsi que n'importe quelle fonction de construction ou tableau de ces types. Il va être vérifié si une prop est de l'un de ces types et un avertissement sera fourni s'il n'en est pas un. [Plus d'information](../guide/components-props.html#Types-des-props) sur les types des props.

Reviewer: ```suggestion
    - Le **type** peut être l'un des constructeurs natifs suivants : `String`, `Number`, `Boolean`, `Array`, `Object`, `Date`, `Function`, `Symbol` ainsi que n'importe quelle fonction de construction ou tableau de ces types. Il va être vérifié si une prop est de l'un de ces types et un avertissement sera fourni s'il n'en est pas un. [Plus d'information](../guide/components-props.html#Types-des-props) sur les types des props.
```

Reviewer 2: Remplacer : 

> il va être vérifié si une prop est de l'un de ces types et un avertissement sera fourni s'il n'en est pas un.

par 

> Les types des props seront vérifiés. Un avertissement sera fourni si une prop n'est pas du bon type.

c'est plus simple et compréhensible je trouve

    - **default:** `any`

>     Spécifie la valeur par défaut de la prop. Si la prop n'est pas passé, cette valeur sera utilisée à la place. Les objet ou tableau par défaut doivent être retourné depuis une fonction de fabrique.

Reviewer: ```suggestion
    Spécifie la valeur par défaut de la prop. Si la prop n'est pas passée, cette valeur sera utilisée à la place. Les objets ou tableaux par défaut doivent être retournés depuis une fonction de fabrique.
```

    - **required:** `Boolean`

>     Définie si la prop est requise. Dans un environnement autre que de production, un avertissement sera affiché dans la console si cette valeur est évaluée à vrai mais n'est pas passée.

Reviewer 2: > ... évaluée à vrai et que la prop n'est pas passée.

Sinon on dirait que c'est la valeur de `required` qui n'est pas passée

    - **validator:** `Function`

>     Une fonction de validation personnalisée qui prend la valeur de prop comme seul argument. Dans un environnement autre que de production, un avertissement sera affiché dans la console si cette fonction retourne une valeur évaluée à faux. (c.-à-d. que la validation à échouée). Vous pouvez en lire plus à propos de la validation des props [ici](../guide/components-props.html#Validation-des-props).

Reviewer: ```suggestion
    Une fonction de validation personnalisée qui prend la valeur de prop comme seul argument. Dans un environnement autre que de production, un avertissement sera affiché dans la console si cette fonction retourne une valeur évaluée à faux. (c.-à-d. que la validation a échoué). Vous pouvez en lire plus à propos de la validation des props [ici](../guide/components-props.html#Validation-des-props).
```

  Accéder à `vm.$scopedSlots` est surtout utile lors de l'écriture d'un composant avec une [fonction de rendu](../guide/render-function.html).

  **Note :** depuis la 2.6.0+, il y a deux changements notables pour cette propriété :

  1. Les fonctions de slot avec portée garantissent de retourner un tableau de VNodes, sauf si la valeur de retour est invalide, auquel cas la fonction retournera `undefined`.

>   2. Tous les `$slots` sont maintenant exposés via `$scopedSlots` en tant que fonctions. Si vous travaillez avec des fonctions de rendu, il est maintenant recommandé de toujours accéder aux slots via `$scopedSlots`, si elles utilisent actuellement une portée ou non. Cela ne va pas seullement permettre d'ajouter des scopes simplement lors de futures refactorisation, mais également de faciliter votre éventuelle migration vers Vue 3, où tous les slots seront des fonctions.

Reviewer: ```suggestion
  2. Tous les `$slots` sont maintenant exposés via `$scopedSlots` en tant que fonctions. Si vous travaillez avec des fonctions de rendu, il est maintenant recommandé de toujours accéder aux slots via `$scopedSlots`, si elles utilisent actuellement une portée ou non. Cela ne va pas seullement permettre d'ajouter des scopes simplement lors de futures refactorisations, mais également de faciliter votre éventuelle migration vers Vue 3, où tous les slots seront des fonctions.
```

>   2. Tous les `$slots` sont maintenant exposés via `$scopedSlots` en tant que fonctions. Si vous travaillez avec des fonctions de rendu, il est maintenant recommandé de toujours accéder aux slots via `$scopedSlots`, si elles utilisent actuellement une portée ou non. Cela ne va pas seullement permettre d'ajouter des scopes simplement lors de futures refactorisation, mais également de faciliter votre éventuelle migration vers Vue 3, où tous les slots seront des fonctions.

Reviewer 5: ```suggestion
  2. Tous les `$slots` sont maintenant exposés via `$scopedSlots` en tant que fonctions. Si vous travaillez avec des fonctions de rendu, il est maintenant recommandé de toujours accéder aux slots via `$scopedSlots`, si elles utilisent actuellement une portée ou non. Cela ne va pas seulement permettre d'ajouter des scopes simplement lors de futures refactorisations, mais également de faciliter votre éventuelle migration vers Vue 3, où tous les slots seront des fonctions.
```

  </div>
  ```

>   En 2.6+, `v-for` peut également travailler sur des valeurs qui implémente le [protocole Iterable](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Les_protocoles_iteration#The_iterable_protocol), incluant les natifs `Map` et `Set`. Cependant, il est a noté que Vue 2.x ne supporte pour le moment pas la réactivité sur les valeurs de `Map` et `Set` et ne pourra pas automatiquement détecter les changements.

Reviewer: ```suggestion
  En 2.6+, `v-for` peut également travailler sur des valeurs qui implémentent le [protocole Iterable](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Les_protocoles_iteration#The_iterable_protocol), incluant les natifs `Map` et `Set`. Cependant, il est a noté que Vue 2.x ne supporte pour le moment pas la réactivité sur les valeurs de `Map` et `Set` et ne pourra pas automatiquement détecter les changements.
```

>   En 2.6+, `v-for` peut également travailler sur des valeurs qui implémente le [protocole Iterable](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Les_protocoles_iteration#The_iterable_protocol), incluant les natifs `Map` et `Set`. Cependant, il est a noté que Vue 2.x ne supporte pour le moment pas la réactivité sur les valeurs de `Map` et `Set` et ne pourra pas automatiquement détecter les changements.

Reviewer 5: ```suggestion
  En 2.6+, `v-for` peut également travailler sur des valeurs qui implémentent le [protocole Iterable](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Les_protocoles_iteration#The_iterable_protocol), incluant les natifs `Map` et `Set`. Cependant, il est a noter que Vue 2.x ne supporte pour le moment pas la réactivité sur les valeurs de `Map` et `Set` et ne pourra pas automatiquement détecter les changements.
```

>   En 2.6+, `v-for` peut également travailler sur des valeurs qui implémentent le [protocole Iterable](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Les_protocoles_iteration#The_iterable_protocol), incluant les natifs `Map` et `Set`. Cependant, il est a noter que Vue 2.x ne supporte pour le moment pas la réactivité sur les valeurs de `Map` et `Set` et ne pourra pas automatiquement détecter les changements.

Reviewer 3: - il est à noter

  - [Liaisons sur les champs de formulaire](../guide/forms.html)
  - [Composants - Composants de champ de formulaire utilisant les évènements personnalisés](../guide/components.html#Composants-de-champ-de-formulaire-utilisant-les-evenements-personnalises)

### v-slot

- **Notation abrégée :** `#`

> - **Attend comme valeur :** Une expression JavaScript qui est valide a une position d'argument dans une fonction (supporte la décomposition dans les [environments la supportant](../guide/components-slots.html#Décomposition-de-props-de-slot)). Optionnel - seulement nécessaire si le passage des props est attendu par le slot.

Reviewer 3: - environnements

- **Argument :** nom du slot (optionel, `default` par défaut)

- **Limité à :**
  - `<template>`
  - [composants](../guide/components-slots.html#Syntaxe-abrégée-pour-les-slots-par-défault-uniques) (pour les slot par défaut avec props)

- **Utilisation :**

>   Indiquez les slots nommées ou slots qui attendent de recevoir des props.

Reviewer 5: ```suggestion
  Indiquez les slots nommés ou slots qui attendent de recevoir des props.
```

- **Exemple :**

  ```html

>   <!-- Slots nommées -->

Reviewer 5: ```suggestion
  <!-- Slots nommés -->
```

  <base-layout>
    <template v-slot:header>
      Contenu d'en-tête
    </template>

    Contenu par défaut de slot

    <template v-slot:footer>
      Contenu de pied de page
    </template>
  </base-layout>

>   <!-- Slot nommé recevant des props -->

Reviewer: Il faut définir si `slot` est masculin ou féminin (je pense plutôt pour le premier). En effet, tu accordes les adjectifs différemment suivant les lignes.

Author: Je pense tout comme toi le premier ! Merci.

Masculin, sinon c'est une erreur. Comme ici parfois je parle de « portée » (féminin) et « slot » (masculin) je m'embrouille les pinceaux quand je vais trop vite.

  - [Composants dynamiques](../guide/components.html#Composants-dynamiques)
  - [Limitations de l’analyse d’un template à partir du DOM](../guide/components.html#Limitations-de-l’analyse-d’un-template-a-partir-du-DOM)

### slot <sup style="color:#c92222">déprécié</sup>

**Préférez [v-slot](#v-slot) en 2.6.0+.**

- **Attend comme valeur :** `string`

>   Utilisez sur un contenu insérré dans un composant enfant pour indiquer à quel slot nommé le contenu appartient.

Reviewer 5: ```suggestion
  Utilisez sur un contenu inséré dans un composant enfant pour indiquer à quel slot nommé le contenu appartient.
```
